src/zenml/artifacts/utils.py

```diff
@@ -82,20 +82,129 @@
 # ----------
 
 
+def _save_artifact_visualizations(
+    data: Any, materializer: "BaseMaterializer"
+) -> List[ArtifactVisualizationRequest]:
+    """Save artifact visualizations.
+
+    Args:
+        data: The data for which to save the visualizations.
+        materializer: The materializer that should be used to generate and
+            save the visualizations.
+
+    Returns:
+        List of requests for the saved visualizations.
```

Comment:
1. I don't see that return is used in the code
2. If it is used, why we return Request for already existing entity - shall be Response, IMO

Reply:
Unless I'm missing something, the return should be right below?
This function saves the visualizations into the artifact store, and returns the list of requests that reference the paths where they were stored. They get sent to the server as part of the `ArtifactVersionRequest(visualizations=...)`.

Maybe the function name should be different? `_save_artifact_visualizations_to_artifact_store`?

Reply:
You are right the return is used indeed. And I unrolled this logic better from the code and checked how it is used - all good